### PR TITLE
feat: porting over transaction screens

### PR DESCRIPTION
**Pull request by JackShort**
* porting over transaction screens for nfts

### CODE REVIEW

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                      <Box display="flex" flexWrap="wrap" height="min" width="52" key={index}>
+                        <img className={styles.refundAssetImage} src={nft.imageUrl} alt={nft.name} key={index} />
+                      </Box>
+                    ))}
+                  </Box>
+                  <Box className={styles.refundOverflowFade} />
+                </Box>
+              ) : (
+                // Only showing when all assets are unavailable
+                <Box className={styles.fullRefundModal} onClick={stopPropagation}>
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:252]: The colors are off in dark mode for all unavailable modal, screenshot in slack

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    let totalPurchaseValue = BigNumber.from(0)
+    let totalRefundValue = BigNumber.from(0)
+    let totalUSDRefund = 0
+    let txFeeFiat = 0
+
+    if (transactionResponse !== undefined) {
+      const { nftsPurchased: purchasedNfts, nftsNotPurchased: notPurchasedNfts, txReceipt } = transactionResponse
+      if (nftsPurchased && nftsNotPurchased && txReceipt) {
+        nftsPurchased = purchasedNfts
+        nftsNotPurchased = notPurchasedNfts
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:64]: any reason we do this renaming?

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  const {
+    nftsPurchased,
+    nftsNotPurchased,
+    showPurchasedModal,
+    showRefundModal,
+    totalPurchaseValue,
+    totalRefundValue,
+    totalUSDRefund,
+    txFeeFiat,
+  } = useMemo(() => {
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:50]: this file would be a lot cleaner if we could move this useMemo to a helper function, something like 'parseTransactionResponse'

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    setShowUnavailable(!showUnavailable)
+  }
+
+  function closeTxCompleteScreen() {
+    setTransactionResponse({} as TxResponse)
+    setTxState(TxStateType.New)
+  }
+
+  // Given the length of the array of successfully purchased NFTs, returns the maxHeight and maxWidth of each asset preview
+  const getSuccessfulImageSize = (numSuccessful: number) => {
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:99]: we could move this to the new transaction utils file

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+  useEffect(() => {
+    useSendTransaction.subscribe((state) => (transactionStateRef.current = state.state))
+  }, [])
+
+  return (
+    <>
+      {shouldShowModal && (
+        <Portal>
+          <Overlay onClick={closeTxCompleteScreen} />
+          <Box className={styles.modalContainer} onClick={closeTxCompleteScreen}>
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:129]: Mixed success modal cuts into header, screenshot in slack

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+                  <p className={styles.totalEthCost}>
+                    for {nftsNotPurchased.length} unavailable item
+                    {nftsNotPurchased.length === 1 ? '' : 's'}.
+                  </p>
+                  <Box
+                    as="button"
+                    border="none"
+                    backgroundColor="genieBlue"
+                    cursor="pointer"
+                    className={`${styles.returnButton}`}
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:347]: don't need the formatted string here ``

File: src/nft/components/collection/TransactionCompleteModal.tsx [hunk 7]
```diff
@@ -0,0 +1,363 @@
...
+    fetchPrice().then((price) => {
+      setEthPrice(price ?? 0)
+    })
+  }, [])
+
+  useEffect(() => {
+    useSendTransaction.subscribe((state) => (transactionStateRef.current = state.state))
+  }, [])
+
+  return (
```
Comment (cbachmeier) [inline src/nft/components/collection/TransactionCompleteModal.tsx:124]: non-blocking but we could make this file significantly more readable if we split the modal components below into 'SuccessModal', 'MixedRefundModal', and 'FullRefundModal'

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/nft/components/collection/TransactionCompleteModal.tsx (hunk 2 above)
Comment (JackShort) [reply src/nft/components/collection/TransactionCompleteModal.tsx:64]: if its undefined it doesn't get set and an empty array is returned, if it is undefined it gets set and is returned

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**